frontend/src/helpers/user-service/api-wrapper.ts

```diff
@@ -0,0 +1,92 @@
+import { UseUserContext } from "@/contexts/user-context";
+import axios from "axios";
+import Cookies from 'js-cookie';
+import { UserInfo } from "../../components/common/ReadOnlyUserCard";
+
+const api = axios.create({
+  baseURL:
+    process.env.USER_API_ENDPOINT ?? "http://localhost:3001/user",
+  timeout: 5000,
+  headers: {
+    "Content-type": "application/json",
+  },
+});
+
+const login = async (email: string, password: string) => {
+    const { setUser } = UseUserContext();
+
+    await api.post(
+        `/login`,
+        {
+            email: email,
+            password: password
+        },
+        { withCredentials: true}
+    ).then((res) => {
+        if (res.status == 200) {
+            const user = res.data as User;
+            setUser(user);
+            Cookies.set('user', JSON.stringify(user), {expires: 7});
+            return user;
+        } else {
+            throw new Error("Invalid Email/Password");
+        }
+    }).catch((err: Error) => {
+        console.log(err);
+        return {
+            ok: false,
+            status: 500
+        }
+    });
+};
+
+const register = async (email: string, password: string) => {
+    await api.post(
+        `/register`,
+        {
+            email: email,
+            password: password,
+            name: 'name placeholder',
+            major: 'major placeholder',
+            course: 'course placeholder',
+            role: 'role placeholder'
```

Comment:
I think you should change this to "student" as we agreed ytd, unless you have a new idea of implementation?

Reply:
Resolved